coffee_core/src/coffee.rs

```diff
@@ -302,6 +327,23 @@ impl PluginManager for CoffeeManager {
                         } else {
                             self.start_plugin(&path).await?;
                         }
+
+                        if branch.is_some() {
+                            // we can safely unwrap here because we have checked it before
+                            let current_git_head = current_git_head.unwrap();
+                            log::debug!("checking out to git head: {current_git_head}");
+                            // rollback to the previous git head
+                            let updated_status = git_upgrade_with_git_head(
+                                repo.url().path_string.as_str(),
+                                &current_git_head,
+                                verbose,
+                            )
+                            .await?;
+                            log::debug!("updated status: {:?}", updated_status);
+
+                            plugin.commit = current_plugin_commit;
+                        }
```

Comment:
wow this install method is starting to be 2 pages long?

Reply:
I agree!
We can keep `coffee install`  for just installing and configuring plugins and introduce `coffee upgrade --plugin <plugin_name> --branch <branch_name>` for upgrades and branch switching

Reply:
No, my point is that the code that we are writing the code of the install is not good at all. Please see some of my PR that I am refactoring it.

In addition, both install and upgrade command will run the same logic and with your code we should duplicate the checkout logic. This is error prone